cupy/_core/_dtype.pyx

```diff
@@ -57,8 +55,6 @@ cpdef get_dtype(t):
 
 @cython.profile(False)
 cpdef tuple get_dtype_with_itemsize(t):
-    if type(t) is _dtype:  # Exact type check
-        return t, t.itemsize
```

Comment:
The exact check never works, `type(t)` is always a subclass.  I doubt this function is used much with instances though, so just deleted it.

Reply:
This fast path is introduced for performance but has not worked since NumPy 1.20. I agree to delete it!